lib/bolt/applicator.rb

```diff
@@ -232,35 +176,32 @@ def count_statements(ast)
     def apply_ast(raw_ast, targets, options, plan_vars = {})
       ast = Puppet::Pops::Serialization::ToDataConverter.convert(raw_ast, rich_data: true, symbol_to_string: true)
 
-      # rubocop:disable Style/GlobalVars
-      if $future
-        # Serialize as pcore for *Result* objects
-        plan_vars = Puppet::Pops::Serialization::ToDataConverter.convert(plan_vars,
-                                                                         rich_data: true,
-                                                                         symbol_as_string: true,
-                                                                         type_by_reference: true,
-                                                                         local_reference: false)
-        scope = {
-          code_ast: ast,
-          modulepath: @modulepath,
-          pdb_config: @pdb_client.config.to_hash,
-          hiera_config: @hiera_config,
-          plan_vars: plan_vars,
-          # This data isn't available on the target config hash
-          config: @inventory.config.transport_data_get
-        }
-      end
-      # rubocop:enable Style/GlobalVars
+      # Serialize as pcore for *Result* objects
+      plan_vars = Puppet::Pops::Serialization::ToDataConverter.convert(plan_vars,
+                                                                       rich_data: true,
+                                                                       symbol_as_string: true,
+                                                                       type_by_reference: true,
+                                                                       local_reference: false)
+
+      pdb_client_hash = @pdb_client.respond_to?(:config) ? @pdb_client.config.to_hash : {}
```

Comment:
Under what circumstance does this not respond to `config`?

Reply:
There's a [test](https://github.com/puppetlabs/bolt/blob/master/spec/bolt_spec/plan_spec.rb#L198-L203) that uses a `BoltSpec::Plans::MockPuppetDBClient` client defined [here](https://github.com/puppetlabs/bolt/blob/master/lib/bolt_spec/plans.rb#L188). When that gets passed to the applicator it errors since `config` returns `nil`. I tried allowing instances of that class to accept `config` and return an empty hash, but the test still failed. I also tried defining a `config` method that returned an empty hash, but it caused a ton of other tests to fail.

Not really sure what the best approach is here... I made this change to at least get the test passing / review started.

Reply:
I think the way to go is to give `MockPuppetDBClient` an initialize method that takes a config, add an `attr_reader :config`, and pass `Bolt::PuppetDB::Config.new({})` when we construct it in `lib/bolt_spec/plans.rb`. Then it ought to behave the way the rest of the code expects.

Defining `config` to return an empty hash _should_ have worked (since `Hash#to_hash` is defined), so if this approach (adding a proper config attribute) causes problems with other tests, it's probably an underlying issue with the test.